lib/pages/chat_details/chat_details_page_view/files/chat_details_files_item_web/chat_details_files_item_web.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:async';
+
+import 'package:dartz/dartz.dart' hide State, OpenFile;
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/pages/chat_details/chat_details_page_view/files/chat_details_files_item_web/chat_details_files_item_view_web.dart';
+import 'package:fluffychat/presentation/model/chat/downloading_state_presentation_model.dart';
+import 'package:fluffychat/utils/manager/download_manager/download_file_state.dart';
+import 'package:fluffychat/utils/manager/download_manager/download_manager.dart';
+import 'package:fluffychat/widgets/mixins/handle_download_and_preview_file_mixin.dart';
+import 'package:fluffychat/widgets/twake_app.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatDetailsFileItemWeb extends StatefulWidget {
+  const ChatDetailsFileItemWeb({super.key, required this.event});
+
+  final Event event;
+
+  @override
+  State<ChatDetailsFileItemWeb> createState() => ChatDetailsFileItemWebState();
+}
+
+class ChatDetailsFileItemWebState extends State<ChatDetailsFileItemWeb>
+    with HandleDownloadAndPreviewFileMixin {
+  final downloadManager = getIt.get<DownloadManager>();
+
+  final downloadFileStateNotifier = ValueNotifier<DownloadPresentationState>(
+    const NotDownloadPresentationState(),
+  );
+
+  StreamSubscription<Either<Failure, Success>>? streamSubscription;
+
+  Event get event => widget.event;
+
+  @override
+  void initState() {
+    super.initState();
+    trySetupDownloadingStreamSubcription();
+    if (streamSubscription != null) {
+      downloadFileStateNotifier.value = const DownloadingPresentationState();
+    }
+  }
+
+  void trySetupDownloadingStreamSubcription() {
+    streamSubscription = downloadManager
+        .getDownloadStateStream(widget.event.eventId)
+        ?.listen(setupDownloadingProcess);
+  }
+
+  void setupDownloadingProcess(Either<Failure, Success> event) {
+    event.fold(
```

Comment:
i see this logic before why don't you put it in the mixin also ?

Reply:
I thought this would make too many changes on one time but I think you're right, change it is better